tests/distributed/test_join_local_from_remote/test_integration.py

```diff
@@ -30,21 +31,22 @@ def client():
 @pytest.mark.timeout(360)
 @pytest.mark.parametrize('docker_compose', [compose_yml], indirect=['docker_compose'])
 def test_flow(docker_compose, doc_to_index, client, mocker):
-    m = mocker.Mock()
 
     def validate_resp(resp):
-        m()
         assert len(resp.search.docs) == 1
         assert resp.search.docs[0].text == 'test'
 
+    mock = mocker.Mock()
     flow_id = create_flow_2(flow_yaml=flow_yaml)
 
-    client.search(input_fn=[doc_to_index], on_done=validate_resp)
+    client.search(input_fn=[doc_to_index], on_done=mock)
 
     assert_request(method='get',
                    url=f'http://localhost:8000/flows/{flow_id}')
 
     assert_request(method='delete',
                    url=f'http://localhost:8000/flows/{flow_id}',
                    payload={'workspace': False})
-    m.assert_called_once()
+
+    mock.assert_called_once()
+    validate_callback(mock, validate_resp)
```

Comment:
what about providing the validate_callback function an additional parameter to assert how often it is called? `validate_callback(mock, validate_resp, calls)`

Reply:
I thought about this back and forth. Had even implemented it three times and removed it again. Reasoning: Checking the amount of calls is a thing on mock via `mock.assert_called_once()` or `assert mock.call_count == `. I did not wanted to put tooo much magic in this function, but only, that it guarantees, the validate will be called with at least one set of parameters. Anyhow, I can see writing our own standardized mock usability function would create benefit in similar usage across the code base. Would adding it as a follow-up PR make sense?